src/components/Home/News/NewsBottomSheet.tsx

```diff
@@ -84,11 +84,22 @@ export const NewsBottomSheet: FC<NewsBottomSheetProps> = props => {
 
   useEffect(() => {
     // Set up the event listener to close the NewsBottomSheet
-    // when the home button in the NavBar is clicked
+    // when the home button in the NavBar or hardware button are clicked
     const listener = newsSheetEventEmitter.addListener("should_close", () => {
       setIsNewsClosed(true)
     })
-    return () => listener.remove?.()
+    const backHandler = BackHandler.addEventListener(
+      "hardwareBackPress",
+      () => {
+        setIsNewsClosed(true)
+        return true
+      }
+    )
+
```

Comment:
Va molto bene, l'unica cosa è che penso sia meglio lasciare il comportamento di default quando `isNewsClosed === true`.

In modo che premendo il tasto back dalla home con le news chiuse si esca dall'applicazione, come accade in tutte le app, cosa ne pensi?

Ho visto che sembra essere necessario utilizzare un useRef,altrimenti isNewsClosed non riflette i cambiamenti dentro il callback.

Circa cosi:

```
  const [isNewsClosed, setIsNewsClosed] = useState(false)
  const isNewsClosedRef = useRef(isNewsClosed)
  // ...

useEffect(() => {
    // dispatch news sheet event
    newsSheetEventEmitter.emit("state_change", !isNewsClosed)
    isNewsClosedRef.current = isNewsClosed
  }, [isNewsClosed])

  //...

  // dentro il callback
   if (isNewsClosedRef.current == false) {
          setIsNewsClosed(true)
          return true
        }
        return false   
```